src/nprintml/util.py

```diff
@@ -50,9 +52,32 @@ def __iter__(self):
     def __repr__(self):
         return f'({self.__class__.__name__}: {self.iterator!r})'
 
+
 prime_iterator = PrimedIterator.prime
 
 
+class ResultsIterator:
+
+    @classmethod
+    def storeresults(cls, generator):
+        @functools.wraps(generator)
+        def wrapped(*args, **kwargs):
+            iterator = generator(*args, **kwargs)
+            return cls(iterator)
+
+        return wrapped
+
+    def __init__(self, iterator):
+        self.iterator = iterator
+        self.result = None
+
+    def __iter__(self):
+        self.result = yield from self.iterator
```

Comment:
(The `storeresults` wrapper. …For whatever reason, modern Python offers this feature, but hasn't brought it as high-level as this helper makes it.)